.github/workflows/build-timestamped-master.yml

```diff
@@ -0,0 +1,57 @@
+name: Build
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}
+  cancel-in-progress: true
+
+on:
+  push:
+      branches:
+          - main
+  workflow_dispatch:
+
+jobs:
+  ubuntu-build:
+      name: Build
+      runs-on: ubuntu-latest
+      concurrency:
+        group: ${{ github.head_ref }}-ubuntu-build
+        cancel-in-progress: true
+      steps:
+          - name: Checkout Repository
+            uses: actions/checkout@v3
+
+          - name: Set Up Ballerina
+            uses: ballerina-platform/setup-ballerina@v1.1.0
+            with:
+              version: latest
+              
+          - name: Set up JDK 17
+            uses: actions/setup-java@v3
+            with:
+                distribution: 'temurin'
+                java-version: 17.0.7
+          - name: Change to Timestamped Version
+            run: |
+              startTime=$(TZ="Asia/Kolkata" date +'%Y%m%d-%H%M00')
+              latestCommit=$(git log -n 1 --pretty=format:"%h")
+              VERSION=$((grep -w 'version' | cut -d= -f2) < gradle.properties | rev | cut --complement -d- -f1 | rev)
+              updatedVersion=$VERSION-$startTime-$latestCommit
+              echo $updatedVersion
+              sed -i "s/version=\(.*\)/version=$updatedVersion/g" gradle.properties
+          - name: Set ENV Variables
+            run: |
+                echo -e '${{ toJson(secrets) }}' | jq -r 'to_entries[] | .key + "=" + .value' >> $GITHUB_ENV 
+
+          - name: Grant execute permission for gradlew
+            run: chmod +x gradlew     
+
+          - name: Build the Package
+            env:
+              packageUser: ${{ github.actor }}
+              packagePAT: ${{ secrets.GITHUB_TOKEN }}
+            run: |
+                ./gradlew fullbuild
```

Comment:
Shall we use standard task names? `./gradlew build`

Reply:
If a developer has previously installed EDI tool, the `./gradlew build` command already available for building the package. This means the developer should already have the `bal-tools.toml `file in their `.ballerina/.config/` directory.

On the other hand, `./gradlew fullbuild` is intended for running PR and Release workflows that create a clean `bal-tools.toml` in the `.ballerina/.config/` directory on GitHub's VM.

The reason to having a separate task for that is, if we create a clean `bal-tools.toml`(with only allowing EDI Tool) in developer's `.ballerina/.config/` directory, the existing tools that he used other than EDI tool will be unusable.